src/parquet/metadata.cc

```diff
@@ -124,8 +155,13 @@ class ColumnChunkMetaData::ColumnChunkMetaDataImpl {
   //    Eg: UTF8
   inline bool is_stats_set() const {
     DCHECK(writer_version_ != nullptr);
-    return column_->meta_data.__isset.statistics &&
-           writer_version_->HasCorrectStatistics(type(), descr_->sort_order());
+    if (!column_->meta_data.__isset.statistics) {
+      return false;
+    }
+    std::unique_ptr<MinMaxStatistics> statistics = MinMaxStatistics::Make(
+            reinterpret_cast<const uint8_t*>(&column_->meta_data.statistics));
```

Comment:
You need to use `TypedRowGroupStatistics` to get type based equality. You can simply solve your problem by creating that object (via `MakeColumnStats()`) here.
You will have to re-organize `is_stats_set` and `statistics()` to avoid calling `MakeColumnStats()` twice.

Reply:
@majetideepak So I implemented that but that complicated things since for the new statistics it always says min/max is set (see https://github.com/apache/parquet-cpp/blob/master/src/parquet/metadata.cc#L57)

This makes it seem like min and max are equivalent when they are not set (since they will both be empty strings)